geo/geodataset.cpp

```diff
@@ -722,6 +722,35 @@ GDALResampleAlg algoToGdal(GeoDataset::Resampling alg)
     return GRA_Lanczos;
 }
 
+const std::string algoToGdalString(GeoDataset::Resampling alg)
+{
+    switch (alg) {
+    case GeoDataset::Resampling::nearest: return "NEAREST";
+    case GeoDataset::Resampling::bilinear: return "BILINEAR";
+    case GeoDataset::Resampling::cubic: return "CUBIC";
+    case GeoDataset::Resampling::cubicspline: return "CubicSpline";
+    case GeoDataset::Resampling::lanczos: return "Lanczos";
+    case GeoDataset::Resampling::average: return "Average";
+    case GeoDataset::Resampling::mode: return "Mode";
+
+    case GeoDataset::Resampling::minimum:
+    case GeoDataset::Resampling::maximum:
+    case GeoDataset::Resampling::median:
+    case GeoDataset::Resampling::q1:
+    case GeoDataset::Resampling::q3:
+        LOGTHROW( err2, std::runtime_error )
+            << "Resampling method " << alg
+            << " has no GDAL string equivalent.";
+        break;
+
+    case GeoDataset::Resampling::texture: return "AVERAGE";
```

Comment:
Is this different to `"Average"`?

Reply:
It seems gdal uses case-insensitive matching. Would be nice to have everything in upper case, as is used in the doc.